src/main/scala/is/hail/utils/richUtils/RichIndexedRowMatrix.scala

```diff
@@ -1,66 +1,90 @@
 package is.hail.utils.richUtils
 
-import org.apache.spark.mllib.linalg.{DenseMatrix, Matrix}
-import org.apache.spark.mllib.linalg.distributed.{BlockMatrix, IndexedRowMatrix}
-import org.apache.spark.rdd.RDD
-
+import breeze.linalg.{DenseMatrix => BDM}
+import is.hail.distributedmatrix._
 import is.hail.utils._
+import org.apache.spark.mllib.linalg.distributed.IndexedRowMatrix
 
-/**
-  * Adds toBlockMatrixDense method to IndexedRowMatrix
-  *
-  * The hope is that this class is temporary, as I'm going to make a PR to Spark with this method
-  * since it seems broadly useful.
-  */
+object RichIndexedRowMatrix {
+  private def seqOp(truncatedBlockRow: Int, truncatedBlockCol: Int, excessRows: Int, excessCols: Int, blockSize: Int)
+    (block: Array[Double], row: (Int, Int, Int, Array[Double])): Array[Double] = row match {
+    case (i, j, ii, a) =>
+      val rowsInBlock: Int = if (i == truncatedBlockRow) excessRows else blockSize
+      val colsInBlock: Int = if (j == truncatedBlockCol) excessCols else blockSize
+      val block2 = if (block == null) new Array[Double](rowsInBlock * colsInBlock) else block
 
-class RichIndexedRowMatrix(indexedRowMatrix: IndexedRowMatrix) {
-  def toBlockMatrixDense(): BlockMatrix = {
-    toBlockMatrixDense(1024, 1024)
+      var jj = 0
+      while (jj < a.length) {
+        block2(jj * rowsInBlock + ii) = a(jj)
```

Comment:
Not necessary in this PR, but we may want to eventually add `toTransposedHailBlockMatrix()` as well (which directly produced the transpose, with `isTranspose = false`) which looks like it entails modifying these lines and the blockRow and blockCol indices. I'm thinking of a case like `irm.toHailBlockMatrix().t * otherBlockMatrix` where it may be faster for the conversion (due to stride) and the multiplication (due to the BDM blocks not actually be transposed) to have `irm.toTransposedHailBlockMatrix() * otherBlockMatrix`. As written, the BlockMatrix ops will in general perform best when all BlockMatrices have the same transposition state at the first one. In any case, we'll first want to understand better where time is being spent and have a clear use case (I need to look closer at LMM).